workers/data_refinery_workers/processors/transcriptome_index.py

```diff
@@ -0,0 +1,230 @@
+from __future__ import absolute_import, unicode_literals
+import os
+import subprocess
+import tarfile
+import gzip
+import shutil
+from typing import Dict
+from celery import shared_task
+from data_refinery_common.models import File, BatchKeyValue
+from data_refinery_workers.processors import utils
+from data_refinery_common.logging import get_and_configure_logger
+
+
+logger = get_and_configure_logger(__name__)
+
+
+JOB_DIR_PREFIX = "processor_job_"
+GENE_TO_TRANSCRIPT_TEMPLATE = "{gene_id}\t{transcript_id}\n"
+GENE_TYPE_COLUMN = 2
+# Removes each occurrance of ; and "
+IDS_CLEANUP_TABLE = str.maketrans({";": None, "\"": None})
+
+
+def _set_job_prefix(job_context: Dict) -> str:
+    job_context["job_dir_prefix"] = JOB_DIR_PREFIX + str(job_context["job_id"])
+    return job_context
+
+
+def _prepare_files(job_context: Dict) -> Dict:
+    """Moves the batch's files from the raw directory to the temp directory.
+
+    Also adds the keys "fasta_file_path" and "gtf_file_path" to
+    job_context.
+    """
+    # Transcriptome index processor jobs have only one batch. Each
+    # batch has a fasta file and a gtf file
+    batch = job_context["batches"][0]
+    fasta_file = File.objects.get(batch=batch, raw_format__exact="fa.gz")
+    gtf_file = File.objects.get(batch=batch, raw_format__exact="gtf.gz")
+
+    try:
+        fasta_file.download_raw_file(job_context["job_dir_prefix"])
+        gtf_file.download_raw_file(job_context["job_dir_prefix"])
+    except Exception:
+        logger.exception("Exception caught while retrieving raw files.",
+                         processor_job=job_context["job_id"],
+                         batch=batch.id)
+
+        job_context["job"].failure_reason = "Exception caught while retrieving raw files."
+        job_context["success"] = False
+        return job_context
+
+    # The files are gzipped when download, but rsem-prepare-reference
+    # expects them to be unzipped.
+    gzipped_fasta_file_path = fasta_file.get_temp_pre_path(job_context["job_dir_prefix"])
+    gzipped_gtf_file_path = gtf_file.get_temp_pre_path(job_context["job_dir_prefix"])
+    job_context["fasta_file_path"] = gzipped_fasta_file_path.replace(".gz", "")
+    job_context["gtf_file_path"] = gzipped_gtf_file_path.replace(".gz", "")
+
+    with gzip.open(gzipped_fasta_file_path, "rb") as gzipped_file, \
+            open(job_context["fasta_file_path"], "wb") as gunzipped_file:
+        shutil.copyfileobj(gzipped_file, gunzipped_file)
+
+    with gzip.open(gzipped_gtf_file_path, "rb") as gzipped_file, \
+            open(job_context["gtf_file_path"], "wb") as gunzipped_file:
+        shutil.copyfileobj(gzipped_file, gunzipped_file)
+
+    job_context["fasta_file"] = fasta_file
+    job_context["gtf_file"] = gtf_file
+    return job_context
```

Comment:
Do you need to return `job_context` too when no exception is raised?

Reply:
Yes, each step of a pipeline must return a `job_context` because that will be fed into the next step. See https://github.com/data-refinery/data_refinery/blob/master/workers/data_refinery_workers/processors/utils.py#L143